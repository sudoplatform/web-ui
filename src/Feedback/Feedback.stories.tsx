import { storiesOf } from '@storybook/react'
import React from 'react'

import { Feedback } from './Feedback'

storiesOf('Feedback', module)
  .add('default', () => <Feedback>Default (Info)</Feedback>)
  .add('success', () => <Feedback type="success">Success</Feedback>)
  .add('warning', () => <Feedback type="warning">Warning</Feedback>)
  .add('info', () => <Feedback type="info">Info</Feedback>)
  .add('error', () => <Feedback type="error">Error</Feedback>)
