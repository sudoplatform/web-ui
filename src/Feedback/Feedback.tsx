import { Alert } from 'antd'
import Color from 'color'
import React from 'react'
import styled from 'styled-components'

const StyledAlert = styled(Alert)`
  width: 100%;
  border-radius: ${({ theme }) => theme.borderRadius.small};

  &.ant-alert-success {
    border-color: ${({ theme }) =>
      new Color(theme.colors.success).alpha(0.5).string()};
    background: ${({ theme }) =>
      new Color(theme.colors.success).alpha(0.1).string()};
  }

  &.ant-alert-info {
    border-color: ${({ theme }) =>
      new Color(theme.colors.primary).alpha(0.5).string()};
    background: ${({ theme }) =>
      new Color(theme.colors.primary).alpha(0.1).string()};
  }

  &.ant-alert-error {
    border-color: ${({ theme }) =>
      new Color(theme.colors.error).alpha(0.5).string()};
    background: ${({ theme }) =>
      new Color(theme.colors.error).alpha(0.1).string()};
  }
`

interface Props {
  type?: 'success' | 'info' | 'error' | 'warning'
  children?: React.ReactNode
}

export const Feedback: React.FC<Props> = (props) => {
  return <StyledAlert type={props.type} message={props.children} />
}
