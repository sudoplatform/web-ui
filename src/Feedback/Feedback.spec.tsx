import { shallow } from 'enzyme'
import React from 'react'

import { Feedback } from './Feedback'

describe('Feedback', () => {
  it('should render', () => {
    const wrapper = shallow(<Feedback>TEST</Feedback>)
    expect(wrapper).toMatchSnapshot()
  })
})
