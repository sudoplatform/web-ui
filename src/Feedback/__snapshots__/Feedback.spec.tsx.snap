// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback should render 1`] = `
<Styled(Alert)
  message="TEST"
/>
`;
